### PR TITLE
Filter

### PR DESCRIPTION
**Pull request by partha8**
This PR also includes features like "add a new note" and "pin/unpin a note".
## files to check
- [src/components/FilterModal/FilterModal.js](https://github.com/partha8/melonn/compare/filter?expand=1#diff-5fd1ac7b7da8d2a5c7dac5f7646ce51cc998b14b572afa1f7733d4b9000692f9)
- [src/context/FilterProvider.js](https://github.com/partha8/melonn/compare/filter?expand=1#diff-d8817a799230858d16e3f680a1b1642e9474616471dc0d090cdd86db45400d5d)
- [src/context/NotesProvider.js](https://github.com/partha8/melonn/compare/filter?expand=1#diff-585c9a76075cc76bdae9dea74243ed5b98df79dcaff96225f00c61ce56d04090)
- [src/hooks/useFilteredNotes.js](https://github.com/partha8/melonn/compare/filter?expand=1#diff-7c200d613b3773a91649f18d6ff23e7e91310158fc0f8c30dcf19202b4b36215)
- [src/reducers/filterReducer.js](https://github.com/partha8/melonn/compare/filter?expand=1#diff-9a894a856e4c53d43d41ff13c57f4fa7cf4f4a66cddb1b6572b2b352ba417b3f)
## preview
filter

### CODE REVIEW

File: src/components/EditorContainer.js/EditorContainer.js [hunk 1]
```diff
@@ -37,21 +38,26 @@ export const EditorContainer = () => {
...
-      updateNote(title, body, tag, priority, selectedNoteID);
-    }, 1000);
+    let timer;
+    if (isEditing) {
+      timer = setTimeout(() => {
+        updateNote(title, body, tag, priority, selectedNoteID);
+      }, 1000);
+      setIsEditing(false);
+    }
     return () => clearTimeout(timer);
```
Comment (nayyyhaa) [inline src/components/EditorContainer.js/EditorContainer.js:54]: Appreciation: Great that you are cleaning up timeouts.

File: src/hooks/useFilteredNotes.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { useFilterContext } from "../context";
+
+export const useFilteredNotes = (pinnedNotes, notes) => {
+  let newPinnedNotes = [...pinnedNotes];
+  let newNotes = [...notes];
```
Comment (nayyyhaa) [inline src/hooks/useFilteredNotes.js:5]: Appreciation: Good use of filter utility function in separate file.

**Review by nayyyhaa — APPROVED** [review]
Great work path Parth! Code is nicely written & easily understandable. Found nothing much to suggest. You can merge the changes.

File: src/hooks/useFilteredNotes.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    );
+    newNotes = newNotes.sort((a, b) => b["createdAt"] - a["createdAt"]);
+  }
+  if (sortByTag.length > 0) {
+    newPinnedNotes = newPinnedNotes.filter((note) =>
+      sortByTag.includes(note.tag)
+    );
+    newNotes = newNotes.filter((note) => sortByTag.includes(note.tag));
+  }
+
```
Comment (b0llu) [inline src/hooks/useFilteredNotes.js:44]: Really well written code!

**Review by b0llu — APPROVED** [review]
Great work in PR. Nothing to suggest you can merge this

**Pull request merged**